src/components/portfolio/Details.tsx

```diff
@@ -24,430 +30,104 @@ const useStyles = makeStyles(({ palette }: Theme) =>
+  id: string;
   // function to return to the dashboard
   back: () => void;
 };
 
 // functional component that takes the name of the portfolio and a function to switch back to the dashboard
 // returns the entire details page
-const Details: React.FC<DetailsProps> = ({ back }) => {
+const Details: React.FC<DetailsProps> = ({ token, id, back }) => {
+  const [portfolioDetails, setPortfolioDetails] = React.useState<
+    API.PortfolioDetails | undefined
+  >();
+  const [error, setError] = React.useState<ErrorCode | undefined>(undefined);
+
+  const isMounted = React.useRef(true);
+
+  const fetch = async () => {
+    setError(undefined);
+    try {
+      const details = await API.details(token, id);
+      if (isMounted.current) {
+        setPortfolioDetails(details);
+      }
+    } catch (e) {
+      if (isMounted.current) {
+        setError(e.message);
+      }
+    }
+  };
+
+  React.useEffect(() => {
+    fetch();
+    return () => {
+      isMounted.current = false;
+    };
+    // deps must be empty because the function should only be called on mount.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   const classes = useStyles();
+  const { t } = useTranslation();
 
   return (
     <div>
-      <section className={classes.topBanner}>
+      <div className={classes.topBanner}>
         <DetailsHeader
           back={back}
-          name={mock.name}
-          positions={mock.positions}
-        />
-      </section>
-      <section className={classes.main}>
-        <DetailsMain
-          positionCount={mock.positionCount}
-          value={mock.value}
-          score={mock.score}
-          perf7d={mock.perf7d}
-          perf1y={mock.perf1y}
-          risk={mock.risk}
-          positions={mock.positions}
-          figures={mock.keyFigures}
-          nextDividend={mock.nextDividend}
-          dividendPayoutRatio={mock.dividendPayoutRatio}
-          performance={mockPerformance}
+          name={portfolioDetails?.overview.name}
+          positions={portfolioDetails?.positions}
+          editPositions={async (modifications) => {
+            await API.modify(token, id, modifications);
+            // reload portfolio details after successful edit
+            await fetch();
+          }}
         />
-      </section>
+      </div>
+      {!portfolioDetails && !error && <LinearProgress color="secondary" />}
+      {error && (
+        <Container className={classes.container}>
+          <ErrorMessage
+            error={error}
+            messageKey="portfolio.details.errorMessage"
+            handling={
+              error.startsWith('AUTH')
+                ? {
+                    buttonText: 'error.action.login',
+                    action: async () => {
+                      // TODO: go back to login
+                    },
+                  }
+                : {
+                    buttonText: 'error.action.retry',
+                    action: fetch,
+                  }
+            }
```

Comment:
I personally find the ternary operator harder to read when the statements get larger. I guess that's personal taste though so leave it as you wish.

Reply:
if statements cannot be used here, so the only alternative would be to define this object elsewhere (which would probably not improve readability).